packages/plugins/opentelemetry/src/spans.ts

```diff
@@ -385,6 +395,46 @@ export function setUpstreamFetchResponseAttributes(input: {
   });
 }
 
+export function recordCacheEvent(
+  event: string,
+  payload: OnCacheGetHookEventPayload,
+) {
+  trace.getActiveSpan()?.addEvent('gateway.cache.' + event, {
+    'gateway.cache.key': payload.key,
+    'gateway.cache.ttl': payload.ttl,
+  });
+}
+
+export function recordCacheError(
+  action: 'read' | 'write',
+  error: Error,
+  payload: OnCacheGetHookEventPayload,
+) {
+  trace.getActiveSpan()?.addEvent('gateway.cache.error', {
+    'gateway.cache.key': payload.key,
```

Comment:
good naming! 

nit: can also be `graphql.cache.key` to make it more unique in a full architecture,

Reply:
or `graphql.response_cache.key`